src/modules/InstURL.rb

```diff
@@ -86,24 +87,27 @@ def RewriteCDUrl(url)
     def installInf2Url(extra_dir = "")
       return @installInf2Url unless @installInf2Url.nil?
 
-      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
+      zypp_repo_url = Linuxrc.InstallInf("ZyppRepoURL").to_s
 
-      if parts.empty?
+      if zypp_repo_url.empty?
         # Make it compatible with the current behaviour when
         # install.inf does not exist.
         log.warn "No URL specified through ZyppRepoURL"
-        parts = URL.Parse("cd:///")
+        zypp_repo_url = "cd:/"
       end
 
-      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
+      uri = URI(zypp_repo_url)
```

Comment:
It is a tradeoff between old compatibility custom libraries and the standard Ruby libraries. I don't want to hold us back forever. We are using `URI` in some places anyway: https://github.com/search?utf8=%E2%9C%93&q=uri+org%3Ayast&type=Code&ref=searchresults
So to clarify my original point: I disagree with the change being introduced quietly. If we mention it in the changelog ("use URI.parse on ZyppRepoURL") and we are aware of the (small?) risk then it is OK.

More reference: https://en.opensuse.org/openSUSE:Libzypp_URIs